### PR TITLE
Update and extend release checklist

### PR DESCRIPTION
**Pull request by HenKlei**

### CODE REVIEW

**Comment by HenKlei** [comment]
Maybe some comments/details on sdist and how the distribution to pypi works would be nice.

**Comment by HenKlei** [comment]
Thanks @renefritze, I addressed all your points. Maybe you can take another short look?

**Review by renefritze — APPROVED** [review]
Very nice. Thanks. We should still wait for someone that's not me to have look though :)

File: docs/RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
- 1. [ ] Tag commit.
- 1. [ ] Produce sdist with checked out tag, make sure sdist version is correct.
- 1. [ ] Produce tagged docker with `make VER=CURRENT_TAG && make VER=CURRENT_TAG push`.
- 1. [ ] Push tag.
- 1. [ ] Wait for CI build to finish.
- 1. [ ] Bump/create demo docker.
- 1. [ ] Update homepage.
- 1. [ ] Make a GitHub release. Zenodo is hooked into that.
+ 1. [ ] (Create release branch in `pymor/pymor`. Should have a `.x` as last part of the branch name in contrast
+        to the `RELEASE_TAG`.)
```
Comment (pmli) [inline docs/RELEASE_CHECKLIST.md:8]:
```suggestion
 1. [ ] Create a release branch in `pymor/pymor`. Should have a `.x` as the last part of the branch name in contrast
        to the `RELEASE_TAG`.
```

File: docs/RELEASE_CHECKLIST.md [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
+ 1. [ ] Tag commit in `pymor/pymor` as `RELEASE_TAG`.
+        Use an annotated tag (`git tag -a RELEASE_TAG -m RELEASE_TAG`) with the annotation being `RELEASE_TAG`.
+ 1. [ ] Produce sdist with checked out tag `RELEASE_TAG`, make sure sdist version is correct.
```
Comment (pmli) [inline docs/RELEASE_CHECKLIST.md:11]:
```suggestion
 1. [ ] Run `python setup.py sdist` (on checked out tag `RELEASE_TAG`) and check that it produces `dist/pymor-RELEASE_TAG.tar.gz`.
```

File: docs/RELEASE_CHECKLIST.md [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
+ 1. [ ] Tag commit in `pymor/docker` as `RELEASE_TAG`, make sure to use the commit mentioned in the `.env` in the
+        tagged commit in `pymor/pymor`. Use an annotated tag with the annotation being the version number.
+        For instance:
+```
+source pymor/pymor/.env
+cd pymor/docker
+git checkout "${CI_IMAGE_TAG}"
+git tag -a RELEASE_TAG -m RELEASE_TAG
+```
```
Comment (pmli) [inline docs/RELEASE_CHECKLIST.md:20]:
```suggestion

       ```bash
       source pymor/pymor/.env
       cd pymor/docker
       git checkout "${CI_IMAGE_TAG}"
       git tag -a RELEASE_TAG -m RELEASE_TAG
       ```
```

File: docs/RELEASE_CHECKLIST.md [hunk 4]
```diff
@@ -1,16 +1,33 @@
...
+ 1. [ ] Push tag `RELEASE_TAG` to `pymor/docker`.
+ 1. [ ] Wait for CI build to finish in `pymor/docker`.
+ 1. [ ] Push tag `RELEASE_TAG` to `pymor/pymor`.
+ 1. [ ] Wait for CI build to finish in `pymor/pymor`. The tagged commit will be deployed to pypi automatically,
```
Comment (pmli) [inline docs/RELEASE_CHECKLIST.md:24]:
```suggestion
 1. [ ] Wait for CI build to finish in `pymor/pymor`. The tagged commit will be deployed to PyPI automatically,
```

**Review by pmli — CHANGES_REQUESTED** [review]

File: docs/RELEASE_CHECKLIST.md (hunk 2 above)
Comment (HenKlei) [reply docs/RELEASE_CHECKLIST.md:11]:
```suggestion
 1. [ ] Run `python setup.py sdist` (on checked out tag `RELEASE_TAG`) and check that it
        produces `dist/pymor-RELEASE_TAG.tar.gz`.
```

**Comment by HenKlei** [comment]
Thanks @pmli for the nice improvements. From my side this is good to go now, what do you think @pmli?

File: docs/RELEASE_CHECKLIST.md [hunk 5]
```diff
@@ -1,16 +1,35 @@
...
+       cd pymor/docker
+       git checkout "${CI_IMAGE_TAG}"
+       git tag -a RELEASE_TAG -m RELEASE_TAG
+       ```
+ 1. [ ] Push tag `RELEASE_TAG` to `pymor/docker`.
+ 1. [ ] Wait for CI build to finish in `pymor/docker`.
+ 1. [ ] Push tag `RELEASE_TAG` to `pymor/pymor`.
+ 1. [ ] Wait for CI build to finish in `pymor/pymor`. The tagged commit will be deployed to PyPI automatically,
+        see [the developer docs](https://docs.pymor.org/main/developer_docs.html#stage-deploy).
+ 1. [ ] Bump/create demo docker, i.e. in `pymor/docker` go to the `demo`-folder and copy the subfolder of the last
```
Comment (pmli) [inline docs/RELEASE_CHECKLIST.md:28]:
```suggestion
 1. [ ] Bump/create demo docker, i.e., in `pymor/docker` go to the `demo` folder and copy the subfolder of the last
```

**Review by pmli — CHANGES_REQUESTED** [review]
Just one minor thing, otherwise looks good to me.

**Comment by pmli** [comment]
Thanks for working on this @HenKlei.

**Review by pmli — APPROVED** [review]

**Pull request merged**